engine/src/main/java/org/terasology/recording/EventCatcher.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2018 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.recording;
+
+import org.terasology.audio.events.PlaySoundEvent;
+import org.terasology.entitySystem.event.Event;
+import org.terasology.entitySystem.event.internal.PendingEvent;
+import org.terasology.input.cameraTarget.CameraTargetChangedEvent;
+import org.terasology.input.events.InputEvent;
+import org.terasology.logic.characters.CharacterMoveInputEvent;
+
+/**
+ * This class is responsible for catching the events during a Record and send the desired ones to the RecordedEventStore.
+ */
+public class EventCatcher {
+
+    private long startTime;
+    private long eventCounter;
+
+    public EventCatcher() {
+        this.startTime = System.currentTimeMillis();
+        this.eventCounter = 0;
+    }
+
+    /**
+     * Receives a PendingEvent and add it as a RecordedEvent in the RecordedEventStore if it is an event type that should be
+     * recorded.
+     * @param pe PendingEvent to be checked and added
```

Comment:
Protip: avoid abbreviations unless they're really meaningful. `pendingEvent` is far more meaningful here and not really a burden on character count :-)